fix: only fetch growthbookUri in browser contexts

ojeytonwilliams opened the pull request:
Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

This should stop the build from failing in CD.

The reason it passed in CI is that we don't provide a growthbookUri env var, so fetch was not called during the build process.

<!-- Feel free to add any additional description of changes below this line -->

ojeytonwilliams committed:
fix: only fetch growthbook in browser contexts

raisedadead reviewed (changes requested):
Wrap this in a `try catch`; the current URI that we have for the Growthbook is unstable - it can go down (it's a single instance API and not even load balanced) and doesn't have a guaranteed uptime SLA like the main API.

You can test this by using `example.com` in your local development

ojeytonwilliams committed:
fix: handle invalid growthbookUris

ahmadabdolsaheb commented:
Thank @ojeytonwilliams for the fix.
Here is a nitpick. fetch is being called for every change in the isSignedIn prop. 
It would be great if we could fetch once outside of the component lifecycle.

ojeytonwilliams committed:
fix: update attributes and features as appropriate

This makes one attempt to set the features, but updates the attributes
whenever a relevant part of user has changed.

ojeytonwilliams commented:
@ahmadabdolsaheb thanks for the review. It made me realise we could do a little more. 

By adding the parts of `user` we're interested in to the dependency array it will update the attributes as those change.  Is that what you'd like to happen?  If you're only interested in the initial state of the user, I'll revert those changes.

ahmadabdolsaheb commented:
@ojeytonwilliams, That makes sense.

naomi-lgbt reviewed (approved):
Let's get this in and unblock the pipeline.
